engine/tx_broadcast.go

```diff
@@ -0,0 +1,46 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chainstate"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+)
+
+func broadcastTransaction(ctx context.Context, tx *Transaction) error {
+	client := tx.Client()
+	chainstateSrv := client.Chainstate()
+
+	defer recoverAndLog(tx.Client().Logger())
+
+	unlock, err := newWriteLock(
+		ctx, fmt.Sprintf(lockKeyProcessBroadcastTx, tx.GetID()), client.Cachestore(),
+	)
+	defer unlock()
+	if err != nil {
+		return err
+	}
+
+	txHex := tx.Hex
+	hexFormat := chainstate.RawTx
+	if chainstateSrv.SupportedBroadcastFormats().Contains(chainstate.Ef) {
+		// try to convert to EF, with rawRx as a fallback
+		efHex, ok := ToEfHex(ctx, tx, client)
+		if ok {
+			txHex = efHex
+			hexFormat = chainstate.Ef
+		}
+	}
```

Comment:
```go
if chainstateSrv.SupportedBroadcastFormats().Contains(chainstate.Ef) {
		efHex, ok := ToEfHex(ctx, tx, client)
		if ok {
			txHex = efHex
			hexFormat = chainstate.Ef
		} else {
			tx.Client().Logger().Warn().Str("txID", tx.ID).Msg("failed to convert transaction to EF format, using RawTx as fallback")
		}
	}
```

Reply:
again log unsuccessful conversions

Reply:
I've just moved the function to another file - that's why it appears as "my" changes.
Additionally, I'm not sure whether or not it is necessary to log it here - it exceeds from this task context and I haven't investigate it (yet).

But, FYI - this PR is just a one step of bigger refactorization (remake) of transaction so this code most probably will be rewritten.

Reply:
ok so maybe you need extra jira ticket to change this ??? not sure

Reply:
The tasks slowly appear on JIRA - but it is a brainstorming still on.